src/AstGen.zig

```diff
@@ -1303,80 +1304,129 @@ fn fnProtoExpr(
                 else
                     .param_anytype;
-                _ = try block_scope.addStrTok(tag, param_name, name_token);
-            } else {
+                break :param try block_scope.addStrTok(tag, param_name, name_token);
+            } else param: {
                 const param_type_node = param.type_expr;
                 assert(param_type_node != 0);
                 var param_gz = block_scope.makeSubBlock(scope);
                 defer param_gz.unstack();
-                const param_type = try expr(&param_gz, scope, coerced_type_ri, param_type_node);
+                const param_type = try expr(&param_gz, params_scope, coerced_type_ri, param_type_node);
                 const param_inst_expected: Zir.Inst.Index = @enumFromInt(astgen.instructions.len + 1);
                 _ = try param_gz.addBreakWithSrcNode(.break_inline, param_inst_expected, param_type, param_type_node);
                 const main_tokens = tree.nodes.items(.main_token);
                 const name_token = param.name_token orelse main_tokens[param_type_node];
                 const tag: Zir.Inst.Tag = if (is_comptime) .param_comptime else .param;
                 const param_inst = try block_scope.addParam(&param_gz, tag, name_token, param_name, param.first_doc_comment);
                 assert(param_inst_expected == param_inst);
-            }
+                break :param param_inst.toRef();
+            };
+
+            if (param_name == 0) continue;
+
+            const sub_scope = try astgen.arena.create(Scope.LocalVal);
+            sub_scope.* = .{
+                .parent = params_scope,
+                .gen_zir = &block_scope,
+                .name = param_name,
+                .inst = param_inst,
+                .token_src = param.name_token.?,
+                .id_cat = .@"function prototype parameter",
+            };
+            params_scope = &sub_scope.base;
         }
         break :is_var_args false;
     };
 
+    var align_gz = block_scope.makeSubBlock(params_scope);
+    defer align_gz.unstack();
     const align_ref: Zir.Inst.Ref = if (fn_proto.ast.align_expr == 0) .none else inst: {
-        break :inst try expr(&block_scope, scope, align_ri, fn_proto.ast.align_expr);
+        const inst = try expr(&block_scope, params_scope, coerced_align_ri, fn_proto.ast.align_expr);
+        if (align_gz.instructionsSlice().len == 0) {
+            // In this case we will send a len=0 body which can be encoded more efficiently.
+            break :inst inst;
+        }
+        const param_index: Zir.Inst.Index = @enumFromInt(astgen.instructions.len + 1);
+        _ = try align_gz.addBreakWithSrcNode(.break_inline, param_index, inst, fn_proto.ast.align_expr);
+        break :inst inst;
     };
 
     if (fn_proto.ast.addrspace_expr != 0) {
         return astgen.failNode(fn_proto.ast.addrspace_expr, "addrspace not allowed on function prototypes", .{});
     }
+    var addrspace_gz = block_scope.makeSubBlock(params_scope);
```

Comment:
Unused.

Reply:
> Unused.

It is but without this compiler crashes with 'use of null' in line 11328 in ```AstGen.zig``` in ```addFunc```. There is following doc comment in above this fuction 
```
/// Must be called with the following stack set up:
    ///  * gz (bottom)
    ///  * align_gz
    ///  * addrspace_gz
    ///  * section_gz
    ///  * cc_gz
    ///  * ret_gz
    ///  * body_gz (top)
```

Reply:
Huh, it's strange how it properly checks each of them individually in the else branch but only `ret_gz` in the then branch.

Reply:
Should I rewrite this check in addFn or let it be as it is?